tajo-core/src/main/java/org/apache/tajo/master/GlobalEngine.java

```diff
@@ -155,7 +174,7 @@ public SubmitQueryResponse executeQuery(Session session, String query, boolean i
       if (isJson) {
         planningContext = buildExpressionFromJson(query);
```

Comment:
Honestly, I didn't estimate the exact time. But, using cache will be much faster than parsing huge json queries, I think.